Put CGAP into poetry (C4-34)

netsettler opened the pull request:
This is getting close to ready, so I figured I should put up a PR.

willronchetti commented on .ebextensions/20_packages.config:
```diff
@@ -8,48 +8,66 @@ packages:
-    command: "source /opt/python/run/venv/bin/activate && python deploy/generate_buildout_cfg.py >> /var/log/deploy.log"
-  035_makeclean:
+  0190_mostly_ugprade_pip:
+    command: "source /opt/python/run/venv/bin/activate && pip install --upgrade pip >> /var/log/deploy.log"
```
Have we determined whether or not its necessary to reactivate the venv before each command? Seems like it shouldn't be the case.

netsettler replied:
I did the experiment earlier and probably still have the logs of the results. I think we got an answer but I must have been busy because I did not follow up one way or the other.

netsettler replied:
Also, we can tighten this up later. For now, actually, I prefer to worry more that diff finds minimal differences between cgap and fourfront. If we decide it's possible, we should upgrade both later to hold diffs constant.

netsettler replied:
But I double-checked and it's not sticky line to line. So we really do need this on every line. I guess it calls `bash` separately on each of these 'command' things.

netsettler replied:
No action taken.

willronchetti commented on .ebextensions/20_packages.config:
```diff
@@ -8,48 +8,66 @@ packages:
     leader_only: true
+  0900_restart_apache:
+    command: "sudo service httpd restart"
+  0901_sleep_10_mins_for_debugging:
```
This should be removed.

netsettler replied:
Thanks, yeah, it was supposed to be temporary.

netsettler replied:
Done.

willronchetti commented on .travis.yml:
```diff
@@ -29,13 +29,13 @@ env:
-- ! "if [ -z \"$SNO_STATUS\" ]; then\n    echo \"Snovault branch build for $SNO_BRANCH
-  is failing; exiting build\"\n    travis_terminate\nelse\n    echo \"Snovault branch
-  $SNO_BRANCH is okay with build status: $SNO_STATUS\"\nfi\n"
+# - SNO_PATH="snovault = git https://github.com/$USER/$SNO_REPO.git rev="
```
I'd just get rid of this altogether.

netsettler replied:
Works for me.

netsettler replied:
Done.

willronchetti commented on .travis.yml:
```diff
@@ -46,23 +46,27 @@ install:
-- pip show pytest
+- poetry install
+- make npm-setup
+#- pip install -U zc.buildout setuptools
```
Same deal here.

netsettler replied:
OK

netsettler replied:
Done.

willronchetti commented on Makefile:
```diff
@@ -1,7 +1,54 @@
-clean:
-	rm -rf node_modules eggs parts bin
+clean:  # clear node modules, eggs, npm build stuff 
```
`make info` belongs in this as well I'd say. I can add that if you want, it probably can be copied over directly from FF more or less.

netsettler replied:
I can add it. Seemed a good idea and I've been adding them elsewhere.

netsettler replied:
Done.

willronchetti commented on deploy/generate_production_ini.py:
```diff
@@ -0,0 +1,144 @@
+        if 'ENV_NAME' not in os.environ:
+            raise GenerationError("ENV_NAME is not set.")
+        parser = argparse.ArgumentParser(description=
+            "Generates a product.ini file from a template appropriate for the given environment,"
```
production*

netsettler replied:
Fixed.

willronchetti commented on pyproject.toml:
```diff
@@ -0,0 +1,228 @@
+
+[tool.poetry.dependencies]
+python = ">=3.5,<3.7"
+# apipkg = "1.4"
```
Remove commented out packages

netsettler replied:
OK

netsettler replied:
Done for most. There were a couple I retained for specific reasons, at least temporarily.

willronchetti commented on src/encoded/__init__.py:
```diff
@@ -1,39 +1,34 @@
-import structlog
-import logging
+from dcicutils.env_utils import get_mirror_env_from_context
+# from pyramid.authorization import ACLAuthorizationPolicy
```
Remove commented out imports

netsettler replied:
Yep. I was doing those one step a time to make sure it tested out OK without them. Seems to. They can be removed now.

willronchetti commented on src/encoded/__init__.py:
```diff
@@ -123,11 +118,15 @@ def app_version(config):
 
         config.registry.settings['snovault.app_version'] = version
 
+    # There is GA Config stuff in Fourfront at this point that is missing here. Intentional? -kmp 8-Apr-2020
```
Yes, no GA for CGAP yet

netsettler replied:
OK. Leaving a comment unless you think I shouldn't.

willronchetti commented on src/encoded/__init__.py:
```diff
@@ -230,12 +231,14 @@ def main(global_config, **local_config):
     config.include('.server_defaults')
     config.include('.root')
     config.include('.types')
+    # Fourfront does this. Do we need that here? -kmp 8-Apr-2020
```
Potentially if that source file is present. It may be included through another mechanism though.

netsettler replied:
The source is not present.
`src/encoded/tests/test_batch_download.py` is present, but the only test in it (which did not refer to a `.batch_download.py` file) is commented out.

netsettler replied:
I'll just leave this as-is with the comment.

willronchetti commented on src/encoded/commands/load_data.py:
```diff
@@ -42,6 +45,9 @@ def main():
 
     # do not run on elasticbeanstalk environments unless using --prod flag
     if env and not args.prod:
+        # NOTE: In fourfront, we only care about stopping this in production. Is CGAP intentionally more paranoid? -kmp 8-Apr-2020
```
Yeah, we don't want to do this right now for any env. We can run it manually if need be.

netsettler replied:
OK

netsettler commented:
I think I've addressed all the comments. I don't think any of my actions are controversial so to keep things moving along I am going to go ahead and merge. Will, feel free to check the recent edits and if you see issues we can fix them in a later PR.
